data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeEvaluatorListener.java

```diff
@@ -112,6 +112,18 @@ public void visitErrorNode(ErrorNode node) {
         throw new RuntimeException("Hit error node in the parse tree: " + node.getText());
     }
 
+    private boolean isBooleanOperator(Operator<?> op) {
```

Comment:
I don't think this will scale. We should add this as a method in `Operator` that needs to be implemented. This way, each implementation is sure to have some value for it.